module/zfs/vdev_draid.c

```diff
@@ -0,0 +1,1657 @@
+ * each physical drive at the same offset. This is the minimum allowable
+ * size since it must be possible to store a full 16M block when there is
+ * only a single data column. Next, we defined a "group" to be a set of
+ * sequential disks containing both the parity and data columns. We allow
+ * groups to span multiple rows in order to align any group size to any
+ * number of physical drives. Finally, a "slice" is comprised of the rows
+ * which contains the target number of groups. The permutation mappings
+ * are applied in a round robin fashion to each slice.
+ *
+ * Given n drives in a group (including parity drives) and m physical drives
+ * (not including the spare drives), we can distribute the groups across r
+ * rows without remainder by selecting the least common multiple of n and m
+ * as the number groups; i.e. ngroups = LCM(n, m).
+ *
+ * In the example below, there are 17 physical drives in the configuration,
+ * with two drives worth of spare capacity. The group size is 8, there are
+ * 15 groups and 8 rows per group. If the row chunk size is 16M, the slice
+ * chunk size will be 128M.
+ *
+ *
+ *                              data disks                         spares
+ *    +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+ *    | 5 | 11| 7 | 13| 6 | 9 | 12| 1 | 4 | 2 | 0 | 10| 15| 8 | 3 | 14| 16|
+ * s  |         group 0               |          group 1          |       |
+ * l  |   |           group 2             |        group 3        |       |
+ * i  |       |           group 4             |        group 5    |       |
+ * c  |           |           group 6             |        group 7|       |
+ * e  |               |           group 8             |           |       |
+ * 1  |  group 9          |           group 10            |       |       |
+ *    |      group 11         |           group 12            |   |       |
+ *    |          group 13         |           group 14            |       |
+ *    +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+ *    | 6 | 12| 8 | 14| 7 | 10| 13| 2 | 5 | 3 | 1 | 11| 16| 9 | 4 | 15| 0 |
+ * s  |         group 15              |          group 16         |       |
+ * l  |   |           group 17            |        group 18       |       |
+ * i  |       |           group 19            |        group 20   |       |
+ * c  |           |           group 21            |       group 22|       |
+ * e  |               |           group 23            |           |       |
+ * 2  |  group 24         |           group 25            |       |       |
+ *    |      group 26         |           group 27            |   |       |
+ *    |          group 28         |           group 29            |       |
+ *    +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+ *
+ *    ...
+ *
+ * Notes:
+ * 1. In this configuration the number of groups equals the number of data
+ *    disks. This will occur but it should never be depended upon.
```

Comment:
configuration=example?

"will occur" = is always the case in the current implementation, but please don't rely on it? or "will not be the case in other examples"?

Reply:
"will not the case in other examples", I'll clarify the wording.  The current implementation does not depend on this.